collections/nemo_nlp/nemo_nlp/data/datasets/bert_pretraining.py

```diff
@@ -228,18 +286,46 @@ def mask_ids(self, ids):
           output_mask: list of binary variables which indicate what tokens has
             been masked (to calculate the loss function for these tokens only)
         """
+
+        # Whole-word masking by default, as it gives better performance.
+        cand_indexes = []
+        for (i, id) in enumerate(ids):
+
+            if len(cand_indexes) >= 1 and not self.tokenizer.ids_to_tokens(
+                    [id])[0].startswith('\u2581'):
+                cand_indexes[-1].append(id)
+            else:
+                cand_indexes.append([id])
+
         masked_ids = []
         output_mask = []
-        for i in ids:
-            if random.random() < self.mask_probability:
-                output_mask.append(1)
+        for cand_index in cand_indexes:
+            if (random.random() < self.mask_probability) and \
+                  cand_index[0] != self.tokenizer.special_tokens["[CLS]"] and \
+                  cand_index[0] != self.tokenizer.special_tokens["[SEP]"]:
                 if random.random() < 0.8:
-                    masked_ids.append(self.tokenizer.special_tokens["[MASK]"])
+                    for cand_index_i in cand_index:
```

Comment:
cand_index_i is unused

Reply:
changed it to be 
`for _ in range(cand_index):`